fastlane/lib/fastlane/actions/sync_code_signing.rb

```diff
@@ -45,6 +45,11 @@ def self.define_provisioning_profile_mapping(params)
           env_variable_name = Match::Utils.environment_variable_name_profile_name(app_identifier: app_identifier,
                                                                                             type: Match.profile_type_sym(params[:type]),
                                                                                         platform: params[:platform])
+
+          if params[:platform].to_s == :catalyst.to_s
+            app_identifier = "maccatalyst.#{app_identifier}"
```

Comment:
Not all Catalyst apps are prefixed with `maccatalyst` now that Xcode 11.4 was released. Maybe `match` needs an option to toggle whether or not `DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER` is in use?